1Password/shell-plugins
# new(mongosh): Support mongosh executable for connecting to a MongoDB database

## Overview


This PR introduces support for the `mongosh` executable within the `atlas` shell plugin. It facilitates connecting to a MongoDB database by fetching secrets from a 1Password item.

## Type of change
 


## Related Issue(s)
  

* Relates: https://github.com/1Password/shell-plugins/pull/198/

## How To Test


- Set up a MongoDB database on a host like MongoDB Atlas or AWS DocumentDB.
- Create a 1Password API Credentials item with the following fields and corresponding values: `username`, `password`, `host`, `port` (at this time this is manual because `mongosh` doesn't seem to source from a configuration file or envvars, which we could import from)
- Run `op plugin init mongosh`
- Run `mongosh`, and you should be connected to the MongoDB Shell. If you don't have a host or port, like in the case of MongoDB Atlas, rather a [Connection String](https://www.mongodb.com/docs/manual/reference/connection-string/), you can run `mongosh ConnectionString` (example: `mongosh "mongodb+srv://cluster0.ygauyzj.mongodb.net"`)

## Couple of notes

- I have configured both the executable-specific importer and the default importer to nil, because there aren't any files/envvars to source from. Because of this, `op plugin init mongosh` kind of fails, that there's no importer configured. I am not sure how to handle this, besides asking the user to create the 1Password item manually. Is there a fallback importer flow that accepts the credential fields one by one? 🤔 
- DefaultProvisioner is set to `nil` for now.
- Injected arguments are provisioned immediately after the executable name, rather than at the end. That's because we need to set the index in the shell plugin code, and placing them at 1 seems to work pretty well.

## Changelog
  

Support for mongosh executable, allowing secure connection to a MongoDB Shell for a MongoDB database.
